Source/EventAnalysis/KTIterativeTrackClustering.cc

```diff
@@ -0,0 +1,374 @@
+        {
+            match = false;
+            for (std::vector<KTProcessedTrackData>::iterator newIt = fNewTracks.begin(); newIt != fNewTracks.end(); ++newIt)
+            {
+                if (this->DoTheyMatch(*compIt, *newIt))
+                {
+                    match = true;
+                    KTDEBUG(itclog, "Found matching tracks");
+                    this->CombineTracks(*compIt, *newIt);
+                    break;
+                }
+                // it is possible that the segments that get combined first are not direct neighbors in time
+                // in that case there can be a track segment very close to an already combined track
+                if (this->DoTheyOverlap(*compIt, *newIt))
+                {
+                    match = true;
+                    KTDEBUG(itclog, "Found overlapping tracks");
+                    this->CombineTracks(*compIt, *newIt);
+                    break;   
+                }
+            }
+
+            if (match == false)
+            {
+                KTProcessedTrackData newTrack(*compIt);
+                fNewTracks.push_back(newTrack);
+            }   
+        }
+        return true;
+    }
+
+
+    void KTIterativeTrackClustering::CombineTracks(KTProcessedTrackData& oldTrack, KTProcessedTrackData& newTrack)
+    {
+        if (oldTrack.GetStartTimeInRunC() < newTrack.GetStartTimeInRunC())
+        {
+            newTrack.SetStartTimeInRunC( oldTrack.GetStartTimeInRunC());
+            newTrack.SetStartTimeInAcq( oldTrack.GetStartTimeInAcq());
+            newTrack.SetStartFrequency( oldTrack.GetStartFrequency());
+            newTrack.SetStartTimeInRunCSigma( oldTrack.GetStartTimeInRunCSigma());
+            newTrack.SetStartFrequencySigma( oldTrack.GetStartFrequencySigma());
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+        if (oldTrack.GetEndTimeInRunC() > newTrack.GetEndTimeInRunC())
+        {
+            newTrack.SetEndTimeInRunC( oldTrack.GetEndTimeInRunC());
+            newTrack.SetEndFrequency( oldTrack.GetEndFrequency());
+            newTrack.SetEndTimeInRunCSigma( oldTrack.GetEndTimeInRunCSigma());
+            newTrack.SetEndFrequencySigma( oldTrack.GetEndFrequencySigma());
+            newTrack.SetSlope( (newTrack.GetEndFrequency() - newTrack.GetStartFrequency())/(newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC()));
+
+        }
+
+        newTrack.SetTotalPower( newTrack.GetTotalPower() + oldTrack.GetTotalPower());
+        newTrack.SetTimeLength( newTrack.GetEndTimeInRunC() - newTrack.GetStartTimeInRunC());
+        newTrack.SetSlopeSigma( sqrt(newTrack.GetSlopeSigma()*newTrack.GetSlopeSigma() + oldTrack.GetSlopeSigma()*oldTrack.GetSlopeSigma()));
+    }
+
+
+    bool KTIterativeTrackClustering::DoTheyMatch(KTProcessedTrackData& Track1, KTProcessedTrackData& Track2)
```

Comment:
Variable names to fix: `Track1`, `Track2`, `slope_condition1`, `slope_condition2`, `time_gap_in_line`, `gap_smaller_than_limit`

Minor indentation fix needed on the final return statement